Moved certain methods out of `PaletteManager`

Lehonti opened the pull request:
I think the `PaletteManager` should only be in charge of managing the state of the palette (and also, perhaps, broadcasting the relevant events `PrimaryColorChanged`, `SecondaryColorChanged`, `RecentColorsChanged`), so I:
- Moved the parsing methods out of it, and into `CairoExtensions.ColorType.cs`
- Re-structured `DoKeyPress`, which was checking if the key that was pressed meant that the palette manager should consume the key press event. The check itself was moved out, and only a `SwapColors` method remained

There are other methods (`PopulateSavedPalette`, `PopulateRecentlyUsedColors`, `SaveCurrentPalette`, `SaveRecentlyUsedColors`) that, in my opinion, should be moved out, but they are a bit more involved, so I haven't done that so far. Perhaps I should leave those for another day.

cameronwhite commented on Pinta.Core/Extensions/Cairo/CairoExtensions.ColorType.cs:
```diff
@@ -102,6 +102,35 @@ static string ExpandColorHex (string hex)
+	/// Converts the color to a hex string in the byte order of the ColorBgra struct,
+	/// for backwards compatibility with existing settings.
+	/// </summary>
+	public static string ToBgraHexString (Color color)
```
I'm okay with moving these out into public methods, but would have a couple suggestions:
- There should be a stronger warning about avoiding these for new code (since the ordering is very unexpected). e.g. something like `This should only be used for backwards compatibility with existing settings. New code should use ToHex()`
- Now that they are public, could you also add a simple unit test for these? Just testing a color that has different values for each component, to make sure the byte ordering translates correctly, would be fine

Lehonti commented:
@cameronwhite I addressed your concerns, but now that I created the tests I am thinking that the names of these extracted methods are misleading.

To see it, look at line 140 of `CairoExtensions.ColorType.cs`: I would expect it to be

`Color bgra = new (color.B, color.G, color.R, color.A);`,

but it is

`Color bgra = new (color.A, color.R, color.G, color.B);`

cameronwhite commented:
Yeah it's a bit confusing - the key thing is the comment `byte order of the ColorBgra struct` which is little-endian, introducing the reversal. The original code was actually parsing the hex string as a uint32 (https://github.com/PintaProject/Pinta/pull/1626/files)

Lehonti commented:
Ah, good point about endianness @cameronwhite, I wasn't considering it but now it makes sense. Is there something else to be done, though? The tests already account for how the method actually behaves.
